### PR TITLE
Add 'Span::parent()' and 'Span::source()' to proc_macro API.

### PR DESCRIPTION
**Pull request by SergioBenitez**
As the title suggests: a couple of useful methods for `proc_macro`.

### CODE REVIEW

File: src/libproc_macro/lib.rs [hunk 1]
```diff
@@ -221,6 +221,19 @@ impl Span {
         }
     }
 
+    /// The `Span` for the tokens in the previous macro expansion from which
+    /// `self` was generated from, if any.
+    pub fn parent(&self) -> Option<Span> {
```
Comment (kennytm) [inline src/libproc_macro/lib.rs]: Needs `#[unstable(feature = "proc_macro")]`. (Similar for `source()`)

```
[00:07:40] error: This node does not have a stability attribute
[00:07:40]    --> /checkout/src/libproc_macro/lib.rs:226:5
[00:07:40]     |
[00:07:40] 226 | /     pub fn parent(&self) -> Option<Span> {
[00:07:40] 227 | |         self.0.ctxt().outer().expn_info().map(|i| Span(i.call_site))
[00:07:40] 228 | |     }
[00:07:40]     | |_____^
[00:07:40] 
[00:07:40] error: This node does not have a stability attribute
[00:07:40]    --> /checkout/src/libproc_macro/lib.rs:233:5
[00:07:40]     |
[00:07:40] 233 | /     pub fn source(&self) -> Span {
[00:07:40] 234 | |         Span(self.0.source_callsite())
[00:07:40] 235 | |     }
[00:07:40]     | |_____^
```

**Comment by SergioBenitez** [comment]
Yes, of course. I'll add some ui-tests as well.

**Comment by jseyfried** [comment]
Great!
Alternative naming idea: `parent` -> `source`, `source` -> `original_source`. `parent` sounds a little strange to me w.r.t. spans. I don't have a super strong opinion though.

cc @nrc

**Comment by SergioBenitez** [comment]
Added tests and fixed up the definition. I don't have a strong opinion about naming. `parent` simply felt more natural to me as I'm "moving up the chain".

**Comment by SergioBenitez** [comment]
@jseyfried Look okay?

**Comment by jseyfried** [comment]
Yeah, lets land. @bors r+

**Pull request merged**